.github/workflows/ci.yml

```diff
@@ -109,4 +110,46 @@ jobs:
             echo "No integration tests directory found; skipping."
           fi
 
-
+      - name: Generate coverage report
```

Comment:
looks like the tests run twice? 
- unit tests
- integration tests
- coverage (run both unit and integration)